routes/index.js

```diff
@@ -9,7 +10,57 @@
  */
 module.exports = function routes(app, options, next) {
   app.perform('actions', function (done) {
-    // routes go here
+
+    /**
+     * Returns a route handler for the given type.
+     *
+     * @param {string} type - Event type to get (`Status` or `StatusEvent`)
+     * @returns {function} The async route handler
+     */
+    function statusRouteHandler(type) {
+      const find = (type === 'StatusEvent') ? 'findAll' : 'findOne';
+
+      return async function handler(req, res) {
+        const { pkg, env, version } = req.params;
+
+        if (pkg && env && version) {
+          const status = await app.models[type][find]({ pkg, env, version });
+
+          return res.json(status);
+        }
+
+        if (pkg && env) {
+          const head = await app.models.StatusHead.findOne({ pkg, env });
+          const status = await app.models[type][find]({ pkg, env, version: head.version });
+
+          return res.json(status);
+        }
+
+        res.status(400).json(new Error(`Bad path. ${!pkg && 'Missing package.'} ${!env && 'Missing environment.'}`));
+      };
+    }
+
+    app.routes.get('/status/:pkg/:env/:version?', asynHandler(statusRouteHandler('Status')));
+    app.routes.get('/status-events/:pkg/:env/:version?', asynHandler(statusRouteHandler('StatusEvent')));
+
+    /**
+     * Route Handler for computing progress for the given pkg/env/version
+     *
+     * @param {Request} req - Request object
+     * @param {Response} res - Response object
+     *
+     */
+    app.routes.get('/progress/:pkg/:env/:version?', asynHandler(async function progressHandler(req, res) {
+      let { pkg, env, version } = req.params;
+      let head;
+
+      if (!version) head = await app.models.StatusHead.findOne({ pkg, env });
+
+      version = version || head.version;
+      const progress = await app.progress.compute({ pkg, env, version });
+
+      res.status(200).json({ progress });
```

Comment:
I wonder if it'd be useful to give the percentage and the raw done & total counts.  As it is allows us to add more to this that isn't strictly builds complete, but showing all the data might allow for a richer UI. (e.g. differentiating between 1/3 builds done vs 6/18 builds done).